src/providers/oracle/qgsoracleprovider.cpp

```diff
@@ -1012,6 +964,46 @@ bool QgsOracleProvider::determinePrimaryKey()
   return mValid;
 }
 
+void QgsOracleProvider::determinePrimaryKeyFromUriKeyColumn()
+{
+  QString primaryKey = mUri.keyColumn();
+  mPrimaryKeyType = PktUnknown;
+
+  if ( !primaryKey.isEmpty() )
+  {
+    int idx = fieldNameIndex( primaryKey );
+
+    if ( idx >= 0 )
+    {
+      QgsField fld = mAttributeFields.at( idx );
+
+      if ( mUseEstimatedMetadata || uniqueData( mQuery, primaryKey ) )
+      {
+        mPrimaryKeyType = PktFidMap;
+        if ( fld.type() == QVariant::Int ||
+             fld.type() == QVariant::LongLong ||
+             ( fld.type() == QVariant::Double && fld.precision() == 0 ) )
+        {
+          mPrimaryKeyType = PktInt;
+        }
```

Comment:
Just a cosmetic remark, I find it more clear when there is if/else affectation or ?/: instead of affectation+condition affectation.

Reply:
affectation?

Reply:
I think he suggested to put 982 in an else branch (a suggestion which I would second)

Reply:
Done tahnks for the review

Reply:
Yes, I was meaning assignment. French mistake :smile: